libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java

```diff
@@ -281,4 +401,62 @@ private void logLambdaExecutionError(
         message.values().removeIf(Objects::isNull);
         LOGGER.error(new StringMapMessage(message));
     }
+
+    private void logApiRequestError(PrivateApiResponse failedResponse) {
+        LOGGER.error(LogHelper.buildErrorMessage(FAILED_API_REQUEST, failedResponse.reason()));
+    }
+
+    private HttpResponse<String> sendPostHttpRequest(
+            URI uri, String govukSigninJourneyId, String ipAddress, String payload)
+            throws NonRetryableException {
+        var httpRequestBuilder = buildHttpRequest(uri, govukSigninJourneyId, ipAddress);
+        httpRequestBuilder.POST(HttpRequest.BodyPublishers.ofString(payload));
+
+        return sendHttpRequest(httpRequestBuilder.build());
+    }
+
+    private HttpResponse<String> sendGetHttpRequest(
+            URI uri, String govukSigninJourneyId, String ipAddress) throws NonRetryableException {
+        var httpRequestBuilder = buildHttpRequest(uri, govukSigninJourneyId, ipAddress).GET();
+
+        return sendHttpRequest(httpRequestBuilder.build());
+    }
+
+    private HttpRequest.Builder buildHttpRequest(
+            URI uri, String govukSigninJourneyId, String ipAddress) {
+        var requestBuilder =
+                HttpRequest.newBuilder()
+                        .uri(uri)
+                        .header(GOVUK_SIGNIN_JOURNEY_ID_HEADER, govukSigninJourneyId)
+                        .header(CONTENT_TYPE, ContentType.APPLICATION_JSON.toString())
+                        .header(
+                                X_API_KEY_HEADER,
+                                configService.getApiKeySecret(CIMIT_INTERNAL_API_KEY));
+
+        if (ipAddress != null) {
+            requestBuilder.header(IP_ADDRESS_HEADER, ipAddress);
+        }
+        return requestBuilder;
+    }
+
+    private HttpResponse<String> sendHttpRequest(HttpRequest cimitHttpRequest)
+            throws NonRetryableException {
+        try {
+            LOGGER.info(LogHelper.buildLogMessage("Sending HTTP request to CiMit."));
+            return httpClient.send(cimitHttpRequest, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException e) {
+            throw new NonRetryableException(e);
```

Comment:
I don't think `NonRetryableException` is right here... I think you either want to let the `IOException` propagate up to the handler methods and catch it there, or catch here and consolidate all the CIMIT exceptions into a more general `CimitException`...

(I don't think we ever really need to distinguish `CiPutException` from `CiRetrievalException`, for example)